src/controllers/notifications.js

```diff
@@ -71,6 +73,16 @@ const UnacceptableExpiryError = require('../errors/unacceptable-expiry-error')
 exports.post = function * postNotification () {
   const notification = yield requestUtil.validateBody(this, 'Notification')
   log.debug('Got notification: ' + JSON.stringify(notification))
+
+  const verifySignature = this.config.get('notifications.must_verify')
+  if (verifySignature) {
+    const result = model.verifySignature(notification, this.config)
+    if (!result.valid) {
+      log.warn('Signature verification failed: ' + result.error)
+      throw new UnprocessableEntityError('Notification failed signature verification')
```

Comment:
> My understanding is that notification retries address the situation in which one server is down temporarily or, for whatever reason, the notification didn't reach the recipient, which is not the case here. 

Right, so if the ledger gets a 500 error, something was wrong with the server, it should retry. If it gets a 4XX error, something is wrong with the client (ledger) and it should fix the problem before retrying.